AnkiDroid/src/main/java/com/ichi2/anki/preferences/ControlsSettingsFragment.kt

```diff
@@ -181,3 +212,22 @@ enum class ControlPreferenceScreen(
             PREVIEWER -> PreviewerAction.entries
         }
 }
+
+/**
+ * Highlights a search result with proper lifecycle handling for ControlsSettingsFragment.
+ *
+ * This extension handles the specific case where a tab must be selected before highlighting,
+ * ensuring the fragment is in the appropriate lifecycle states.
+ */
+fun ControlsSettingsFragment.highlightPreferenceWithLifecycle(result: SearchPreferenceResult) {
+    lifecycleScope.launch {
+        withStarted {
+            selectTabForPreference(result.key)
+        }
+        withResumed {
+            view?.post {
+                result.highlight(this as PreferenceFragmentCompat)
+            }
+        }
+    }
```

Comment:
The below worked here. Is there a reason for the use of `withResumed` and `View.post`?
```suggestion
    lifecycleScope.launch {
        withStarted {
            selectTabForPreference(result.key)
            result.highlight(this@highlightPreferenceWithLifecycle)
        }
    }
```

Reply:
The suggested change doesn’t work when the searched preference is inside the Previews tab. The tab gets selected correctly, but the preference itself is not highlighted.